saleor/graphql/product/mutations/products.py

```diff
@@ -436,8 +456,8 @@ class Meta:
         description = "Update public metadata for collection."
         permissions = (ProductPermissions.MANAGE_PRODUCTS,)
         public = True
-        error_type_class = ProductError
-        error_type_field = "product_errors"
+        error_type_class = CollectionError
```

Comment:
All mutations that return collection should wrap it in `ChannelContext`. 
You should add it to all deprecated collection meta mutations in this file.